src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,137 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private int arraySize = 16;
+    private int size = 0;
+    private Node<K, V>[] array;
+
+    public MyHashMap() {
+        array = new Node[arraySize];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == (int) (arraySize * LOAD_FACTOR)) {
+            resizeArray();
+        }
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (array[index] == null) {
+            array[index] = newNode;
+        } else {
+            Node<K, V> previousNode = null;
+            Node<K, V> currentNode = array[index];
+            while (currentNode != null) {
+                if (currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                previousNode = currentNode;
+                currentNode = currentNode.next;
+            }
+            if (previousNode != null) {
+                previousNode.next = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return array[0].value;
+        }
+        int index = getIndex(key);
+        Node<K, V> newNode = array[index];
+        while (newNode != null) {
+            if (newNode.key.equals(key)) {
+                return newNode.value;
+            }
+            newNode = newNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return hashCode() % arraySize;
+    }
+
+    private void resizeArray() {
+        arraySize = arraySize * 2;
+        Node<K, V>[] newArray = new Node[arraySize];
+        transfer(newArray);
+        array = newArray;
+    }
+
+    private void transfer(Node<K, V>[] newArray) {
+        for (int i = 0; i < array.length; i++) {
+            Node<K, V> node = array[i];
+            if (node != null) {
+                array[i] = null;
+                do {
+                    Node<K, V> next = node.next;
+                    int index = getIndex(node.key);
+                    node.next = newArray[index];
+                    newArray[index] = node;
+                    node = next;
+                } while (node != null);
+            }
+        }
+    }
+
+    private void putNullKey(V value) {
+        if (array[0] == null) {
+            array[0] = new Node<>(null, value, null);
+            size++;
+        } else {
+            array[0].value = value;
+        }
+    }
```

Comment:
you don't need a separate method for putting of null key, let's rewrite logic so you could remove it